src/crimson/osd/pg.cc

```diff
@@ -374,6 +388,60 @@ PG::do_delete_work(ceph::os::Transaction &t, ghobject_t _next)
   return {_next, false};
 }
 
+Context *PG::on_clean()
+{
+  // Not needed yet (will be needed for IO unblocking)
+  return nullptr;
+}
+
+void PG::on_active_actmap()
+{
+  logger().debug("{}: {} snap_trimq={}", *this, __func__, snap_trimq);
+  std::ignore = seastar::do_until(
+    [this] { return snap_trimq.empty(); },
+    [this] {
+      const auto to_trim = snap_trimq.range_start();
+      snap_trimq.erase(to_trim);
+      const auto needs_pause = !snap_trimq.empty();
+      return seastar::repeat([to_trim, needs_pause, this] {
+        logger().debug("{}: going to start SnapTrimEvent, to_trim={}",
+                       *this, to_trim);
```

Comment:
`pg->state_set(PG_STATE_SNAPTRIM);`?
By tracking `snaptrim` pg state we will be able to properly test `wait_snap_trimming_complete` when running tests.
AFAIK, there aren't any snptrim specific tests so it might be worth to at least support this.

Reply:
Yup, thanks! I will add a commit here or send a follow-up for it as well as for `PG_STATE_SNAPTRIM_ERROR` and `PG_STATE_SNAPTRIM_WAIT`.